### DISCUSSION TITLE
Cannot get disk space information using dockerized netdata agent

### DISCUSSION THREAD

**Discussion by devspectre** [netdata/netdata#12302] [Q&A]
I installed netdata with Docker and exposed the port so I can access it from localhost(localhost:19999).
I need to get free disk space information so I tried to add diskspace.plugin collector.
https://learn.netdata.cloud/docs/collect/enable-configure
Following the documentation, I could not find that file or any file inside `/etc/netdata` inside the docker container.

I think the documentation is not enough on how to setup additional collectors.

@Ferroin could you please help me to understand how to use netdata to collect disk information of my containers?

**Comment by Ferroin** [comment]
Indeed, the docs are a bit lacking in this case, mostly because of a disconnect in how we handle our Docker containers.

Long-term, we have plans to make this just work.

Shorter term, you can copy the relevant file from under `/usr/lib/netdata/conf.d` inside the container to the required location in `/etc/netdata` in the container, and then edit it there (this is actually all that the `edit-config` script does, it’s just there so that users don’t have to go hunting for the default config files).

**Reply by devspectre** [reply]
Thanks for your reply, @Ferroin 
I am still trying to understand how it works.
So I copied the config file and now it's there and looks like this.
```sh
[global]
    # Default
    memory mode = dbengine

    # Size in Mb of in-memory buffer. Default=32Mb
    page cache size = 32

    # https://learn.netdata.cloud/docs/store/change-metrics-storage#calculate-the-system-resources-ram-disk-space-needed-to-store-metrics
    # 64 GB of storage. Approx 1.4 yrs assuming on 256Mb/day.
    dbengine multihost disk space = 655536
    
[plugin:proc:diskspace]
    remove charts of unmounted disks = yes
    update every = 1
    # check for new mount points every = 15
    # exclude space metrics on paths = /proc/* /sys/* /var/run/user/* /run/user/* /snap/* /var/lib/docker/*
    # exclude space metrics on filesystems = *gvfs *gluster* *s3fs *ipfs *davfs2 *httpfs *sshfs *gdfs *moosefs fusectl autofs
    space usage for all disks = auto
    inodes usage for all disks = auto
```

All I want is to get free disk space of my containers.
I still have no idea how to get that information using netdata agent.
So I tried netdata api on swagger, I tried to use `/api/v1/chart/` endpoint but the result does not seem to be what I wanted.
![image](https://user-images.githubusercontent.com/46239206/156576216-750bfceb-073c-4bbe-96fa-7220479cef1b.png)


Any document or blog you would recommend that can help me, or short guide is also appreicated.

**Reply by devspectre** [reply]
@Ferroin could you please let me know how to get free disk space?

Which endpoint should I use?

**Reply by devspectre** [reply]
@MrZammler I could use your help if you have time for this

**Reply by devspectre** [reply]
@Ferroin any comments?

**Comment by vlvkobal** [comment]
Please check your effective configuration using `http://localhost:19999/netdata.conf`. There are two sections you are interested in
```
[plugins]
	diskspace = yes
```
```
[plugin:proc:diskspace]
	# remove charts of unmounted disks = yes
	# update every = 1
	# check for new mount points every = 15
```
But you should [mount](https://github.com/netdata/netdata/issues/5071#issuecomment-460442980) the volumes you are going to monitor. Otherwise, the Netdata container can't access the information about the filesystems.
